app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,555 @@
+  }
+
+  private void locationSettingsResponseBuilder() {
+    task.addOnCompleteListener(task -> {
+      try {
+        LocationSettingsResponse response = task.getResult(ApiException.class);
+        // All location settings are satisfied. The client can initialize location
+        // requests here.
+
+        createHotspotIntent(ACTION_TURN_ON_AFTER_O);
+
+        //}
+      } catch (ApiException exception) {
+        switch (exception.getStatusCode()) {
+          case LocationSettingsStatusCodes.RESOLUTION_REQUIRED:
+            // Location settings are not satisfied. But could be fixed by showing the
+            // user a dialog.
+            try {
+              // Cast to a resolvable exception.
+              ResolvableApiException resolvable = (ResolvableApiException) exception;
+              // Show the dialog by calling startResolutionForResult(),
+              // and check the result in onActivityResult().
+              resolvable.startResolutionForResult(
+                  ZimHostActivity.this,
+                  LOCATION_SETTINGS_PERMISSION_RESULT);
+            } catch (IntentSender.SendIntentException e) {
+              // Ignore the error.
+            } catch (ClassCastException e) {
+              // Ignore, should be an impossible error.
+            }
+            break;
+          case LocationSettingsStatusCodes.SETTINGS_CHANGE_UNAVAILABLE:
+            // Location settings are not satisfied. However, we have no way to fix the
+            // settings so we won't show the dialog.
+            break;
+          default:
+            break;
+        }
+      }
+    });
+  }
+
+  //Advice user to turn on hotspot manually for API<26
+  private void startHotspotManuallyDialog() {
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+    builder.setPositiveButton(getString(R.string.go_to_settings_label),
+        (dialog, id) -> launchTetheringSettingsScreen());
+
+    builder.setNeutralButton(getString(R.string.hotspot_dialog_neutral_button), (dialog, id) -> {
+
+      progressDialog =
+          ProgressDialog.show(this, getString(R.string.progress_dialog_starting_server), "",
+              true);
+      progressDialog.show();
+
+      startFlowable();
+    });
+
+    builder.setTitle(getString(R.string.hotspot_dialog_title));
+    builder.setMessage(
+        getString(R.string.hotspot_dialog_message)
+    );
+    AlertDialog dialog = builder.create();
+    dialog.show();
+  }
+
+  //Keeps checking if hotspot has been turned using the ip address with an interval of 1 sec
+  //If no ip is found after 15 seconds, dismisses the progress dialog
+  private void startFlowable() {
+    Flowable.fromCallable(() -> ServerUtils.getIp())
+        .retryWhen(error -> error.delay(1, TimeUnit.SECONDS))
+        .timeout(15, TimeUnit.SECONDS)
+        .firstOrError()
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(new SingleObserver<String>() {
```

Comment:
use the subscribe method that takes an onSuccessConsumer and and onErrorConsumer.